agents-api/agents_api/activities/task_steps/prompt_step.py

```diff
@@ -169,6 +169,15 @@ async def prompt_step(context: StepContext) -> StepOutcome:
             formatted_tools.append(tool)
 
     if not is_claude_model:
+        # HOTFIX: for groq calls, litellm expects tool_calls_id not to be in the messages
+        prompt = [
+            {
+                k: v
+                for k, v in message.items()
+                if k not in ["tool_calls", "tool_call_id", "user", "continue_", "name"]
+            }
+            for message in prompt
+        ]
         formatted_tools = None
```

Comment:
- [ ] Add a `# FIXME: Enable formatted_tools` once format-tools PR is merged.
- [ ] Shouldn't we limit the HOTFIX only for groq if that's the only broken provider? Because this strips `tool_call_id` from **all** providers.

Reply:
Done